docs/revive_compiler/differences_yul_translation.md

```diff
@@ -0,0 +1,98 @@
+
+Solidity developers deploying dApps to [`pallet-revive`](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/revive) ought to read and understand this section well.
+
+:::tip
+
+For a high level overview about the differences in the overall runtime architecture, please refer first to the [Differences to Ethereum](../differences_to_eth.md) chapter.
+
+:::
+
+## Deploy code vs. runtime code
+Our contract runtime does not differentiate between runtime code and deploy (constructor) code. Instead, both are emitted into a single PVM contract code blob and live on-chain. Therefor, in EVM termonology, the deploy code equals the runtime code (with the exception that in constructor code, the `codesize` can be larger, since the constructor arguments account to the code size too).
+
+## YUL functions
+The below list contains noteworthy differences in the translation of YUL functions.
+
+:::note
+
+Many functions receive memory buffer offset pointer or size arguments. Since the PVM pointer size is 32 bit, supplying memory offset or buffer size values above `2^32-1` will trap the contract immediatly.
+
+The `solc` compiler ought to always emit valid memory references, so Solidity dApp authors don't need to worry about this unless they deal with low level `assembly` code.
+
+:::
+
+### `mload`, `mstore`, `msize`, `mcopy` (memory related functions) 
+In general, revive preserves the memory layout, meaning low level memory operations are supported. However, a few caveats apply:
+- The EVM linear heap memory is emulated using a fixed byte buffer of 64kb. This implies that the maximum memory a contract can use is limited to 64kbit (on Ethereum, contract memory is capped by gas and therefor variable).
+- Thus, accessing memory offsets larger than the fixed buffer size will trap the contract at runtime with `OutOfBound` erros.
+- The compiler might detect and optimize unused memory reads and writes, leading to a different `msize` compared to what the EVM would see. 
+
+### `calldataload`, `calldatacopy`
+In the constructor code, the offset is ignored and this always returns `0`.
+
+### `codecopy`
+Only supported in constructor code.
+
+### `invalid`
+Traps the contract but does not consume the remaining gas.
+
+### `call`, `delegatecall`, `staticall`
+Calls ignore the supplied gas limit (if any) but forward all remaining resources.
+The reason is multifold, see [Differences to Ethereum](../differences_to_eth.md) (section `Cross-Contract Call Gas Limit`).
+
+Contract authors are strongly advised to [protect against re-entrancy attacks](https://docs.soliditylang.org/en/latest/security-considerations.html#reentrancy).
+
+:::warning
+
+The `solc` compiler supplies a gas stipend of `2300` to calls resulting from `address payable.{send,transfer}`. Which means: The resource limits solc injects for balance transfer calls will be ignored and not protect against re-entrancy attacks.
```

Comment:
ha you mean that in Ethereum you are limited in what you can do here
```
contract Receiver {
    receive() external payable {
        // Limited to 2300 gas, prevent from performing re-entrency attacks
    }
}
```

Reply:
Exactly